core/src/test/java/org/apache/druid/testing/junit/LoggerCaptureRule.java

```diff
@@ -78,12 +77,13 @@ public void clearLogEvents()
   {
     static final String NAME = InMemoryAppender.class.getName();
 
-    private final List<LogEvent> logEvents;
+    // need copy-on-write collection for thread safety of iteration and modification concurrently outside of a critical section
```

Comment:
This comment doesn't achieve its goal: to help somebody to see why the enclosing class needs to be thread-safe. With the current comment, in order to do that, the reader (developer) should invoke "Find usages" IDE action and analyze these usages carefully. This is time-consuming. In essence, there should be concurrent access documentation, as described in https://github.com/code-review-checklists/java-concurrency#justify-document.

This comment is also misleading because there are no critical sections in this code.

Reply:
how about:
logEvents has concurrent iteration and modification in CuratorModuleTest::exitsJvmWhenMaxRetriesExceeded(), needs to be thread safe